src/Types/AsciiStringType.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Types;
+
+use Doctrine\DBAL\ParameterType;
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+
+use function sprintf;
+
+class AsciiStringType extends StringType
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getSQLDeclaration(array $column, AbstractPlatform $platform)
+    {
+        $length = $column['length'] ?? null;
+
+        if (! isset($column['fixed'])) {
+            return sprintf('VARCHAR(%d)', $length);
```

Comment:
The SQL should be generated by the platform, not the type. Otherwise, it's not extensible.

I propose the following:
1. Introduce `AbstractPlatform::getAsciiStringTypeDeclarationSQL()` that would call `AbstractPlatform::getVarcharTypeDeclarationSQL()`.
2. Override the implementation in `SQLServer2012Platform`.